Restore snapshot task ignores boolean "latest" param

## Description

Restoring the latest snapshot via the management REST API fails:

```
curl -u "su:<pwd>" --json '{"latest": true}' localhost:4848/repo/snapshot/restore
```

The endpoint returns a taskId, but the task fails with:

```
com.enonic.xp.repo.impl.elasticsearch.snapshot.SnapshotException: Failed to restore snapshot
Caused by: java.lang.NullPointerException: Cannot invoke "String.indexOf(int)" because "str" is null
        at org.elasticsearch.common.regex.Regex.isSimpleMatchPattern(Regex.java:42)
        at org.elasticsearch.action.admin.cluster.snapshots.get.TransportGetSnapshotsAction.masterOperation(...)
```

## Cause

`SnapshotResource.restore` stores `latest` and `force` as **boolean** properties in the task data (`data.addBoolean(...)`), but the task script `tasks/restore/restore.js` compares them strictly against the **string** `'true'`:

```js
bean.setLatest(params.latest === 'true'); // true === 'true' → false
```

So `latest` is always treated as `false`, the restore proceeds with a `null` snapshot name, and Elasticsearch's get-snapshots call fails with an obscure NPE.

The bug has existed since snapshot/restore REST calls were converted to tasks (#9823, PR #11849). The existing unit tests pass `latest` as a string, which is why they don't catch it.

## Fix

- Read `latest`/`force` strictly as booleans in `restore.js` (`params.latest === true`), matching the `CheckBox` (boolean) fields declared in the task descriptor and what `SnapshotResource` actually submits.
- Align the tests with the real contract by passing booleans instead of strings.
- Fail fast in `SnapshotServiceImpl.doRestore` with a clear message when neither `latest` nor a snapshot name is given, instead of the cryptic ES stack trace.

--- a/modules/app/app-system/src/main/resources/tasks/restore/restore.js
+++ b/modules/app/app-system/src/main/resources/tasks/restore/restore.js
@@ -4,8 +4,8 @@ exports.run = function (params, taskId) {
     var bean = __.newBean('com.enonic.xp.app.system.RestoreTaskHandler');
     bean.setSnapshotName(__.nullOrValue(params.snapshotName));
     bean.setRepositoryId(__.nullOrValue(params.repositoryId));
-    bean.setLatest(params.latest === 'true');
-    bean.setForce(params.force === 'true');
+    bean.setLatest(params.latest === true);
+    bean.setForce(params.force === true);
     bean.setTaskId(__.nullOrValue(taskId));
     bean.execute();
 };

--- a/modules/app/app-system/src/test/resources/test/RestoreTaskHandlerTest.js
+++ b/modules/app/app-system/src/test/resources/test/RestoreTaskHandlerTest.js
@@ -1,11 +1,11 @@
 var system = require('/tasks/restore/restore.js');
 
 exports.restore = function () {
-    let params = {snapshotName: 'my-snapshot', repositoryId: 'my-repo', latest: 'false', force: 'true'};
+    let params = {snapshotName: 'my-snapshot', repositoryId: 'my-repo', latest: false, force: true};
     system.run(params, "task");
 };
 
 exports.restoreLatest = function () {
-    let params = {latest: 'true'};
+    let params = {latest: true};
     system.run(params, "task");
 };

--- a/modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/elasticsearch/snapshot/SnapshotServiceImpl.java
+++ b/modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/elasticsearch/snapshot/SnapshotServiceImpl.java
@@ -135,6 +135,11 @@ public class SnapshotServiceImpl
     {
         final String snapshotName = restoreParams.isLatest() ? determineNameOfLatestSnapshot() : restoreParams.getSnapshotName();
 
+        if ( snapshotName == null || snapshotName.isBlank() )
+        {
+            throw new SnapshotException( "Failed to restore snapshot: Snapshot name is required when 'latest' is not set" );
+        }
+
         validateSnapshot( snapshotName );
 
         final RepositoryId repositoryToRestore = restoreParams.getRepositoryId();

